### ISSUE TITLE
Reindexing an existing index

### ISSUE THREAD

**Issue by seekeramento** [toshi-search/Toshi#870]
**Is your feature request related to a problem? Please describe.**
I am not able to find a way to re-index or even delete an existing index. Once the index is created, I have to assume that the schema will never change.

**Does another search engine have this functionality? Can you describe it's function?**
N/A

**Do you have a specific use case you are trying to solve?**
Create an index for a given schema
Add some documents to your created index.
Update the schema by introducing two additional fields.
You should be able to add documents or editing existing ones with the two newly introduced fields.

**Additional context**
N/A

**Comment by hntd187** [comment]
I don't believe this is possible in tantivy, can tantivy update it's schema?

**Comment by seekeramento** [comment]
In such a case, i it at least possible to delete an entire index? I need this programmatically instead of deleting it from the file system.

**Comment by hntd187** [comment]
So recently a https://github.com/quickwit-oss/tantivy/issues/1211 introduced non-strict schemas which means unknown fields don't error, but that doesn't re-index the index or change the schema, yea, I'd imagine you would have to delete and recreate the index at this point. 

The reason there was no delete endpoint yet was deleting a non-empty directory on windows (and I don't know if this is still the case) was inconsistent so we could get confused if windows for whatever reason decided not to delete the directory. Adding a delete endpoint wouldn't be very difficult it would just be inconsistent, what are you running on?

**Comment by seekeramento** [comment]
I see. The scenario I am having is that I need to create indices for given schema. This works fine until I need to update the schema. There is no update, so I thought of deleting and re-adding, but then realized that also deleting indices is not supported.
